app/src/main/java/ch/epfl/favo/user/UserUtil.java

```diff
@@ -95,22 +116,32 @@ public void deleteAccount() {
    *
    * @param user A user object.
    */
-  public void retrieveUserRegistrationToken(User user) {
-    FirebaseInstanceId.getInstance()
-        .getInstanceId()
-        .addOnCompleteListener(
-            task -> {
-              if (!task.isSuccessful()) {
-                return;
-              }
-              String token = Objects.requireNonNull(task.getResult()).getToken();
-              user.setNotificationId(token);
-
-              Map<String, String> notifMap = new HashMap<String, String>();
-              notifMap.put("notificationId", token);
-              collection.updateDocument(user.getId(), notifMap);
-            });
+  public CompletableFuture retrieveUserRegistrationToken(User user) {
+    FirebaseInstanceId instance = DependencyFactory.getCurrentFirebaseNotificationInstanceId();
+    Task<InstanceIdResult> task = instance.getInstanceId();
+    CompletableFuture<InstanceIdResult> futureIdTask =
+        new TaskToFutureAdapter<>(task).getInstance();
+    return futureIdTask.thenCompose(
+        (instanceIdResult) -> {
+          String token = Objects.requireNonNull(instanceIdResult).getToken();
+          user.setNotificationId(token);
+          Map<String, String> notifMap = new HashMap<>();
+          notifMap.put(User.NOTIFICATION_ID, token);
+          return collection.updateDocument(user.getId(), notifMap);
+        });
   }
+  //        addOnCompleteListener(
+  //            task -> {
+  //              if (!task.isSuccessful()) {
+  //                return;
+  //              }
+  //              String token = Objects.requireNonNull(task.getResult()).getToken();
+  //              user.setNotificationId(token);
+  //
+  //              Map<String, String> notifMap = new HashMap<String, String>();
+  //              notifMap.put("notificationId", token);
+  //              collection.updateDocument(user.getId(), notifMap);
+  //            });
```

Comment:
Is there a reason this is commented out but not deleted?

Reply:
Yeah you're right I forgot to delete it